app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -151,9 +149,9 @@ protected void onCreate(Bundle savedInstanceState) {
                 pendingUrl = intent.getStringExtra(RocketMessagingService.PUSH_OPEN_URL);
             } else {
                 if (Settings.getInstance(this).shouldShowFirstrun()) {
-                    this.mainMediator.showFirstRun();
+                    this.screenNavigator.addFirstRunScreen();
                 } else {
-                    this.mainMediator.showHomeScreen();
+                    this.screenNavigator.popToHomeScreen(false);
```

Comment:
there's nothing in the back stack at the phase, so there's nothing gonna be popped. BTW, BrowserFragment is embedded in MainActivity, so it will never be any transaction that related to BrowserFragment.

Reply:
Oh I mean HomeFragment sorry my bad :P